lib/runners/qemu.nix

```diff
@@ -332,46 +332,16 @@ lib.warnIf (mem == 2048) ''
       ''
         (
           ${writeQmp { execute = "qmp_capabilities"; }}
-          ${writeQmp {
-            execute = "input-send-event";
-            arguments.events = [ {
-              type = "key";
-              data = {
-                down = true;
-                key = {
-                  type = "qcode";
-                  data = "ctrl";
-                };
-              };
-            } {
-              type = "key";
-              data = {
-                down = true;
-                key = {
-                  type = "qcode";
-                  data = "alt";
-                };
-              };
-            } {
-              type = "key";
-              data = {
-                down = true;
-                key = {
-                  type = "qcode";
-                  data = "delete";
-                };
-              };
-            } ];
-          }}
-           # wait for exit
+          ${writeQmp { execute = "system_powerdown"; }}
+          # wait for exit
           cat
         ) | \
         ${pkgs.socat}/bin/socat STDIO UNIX:${socket},shut-none
     ''
     else throw "Cannot shutdown without socket";
 
   setBalloonScript =
-    if socket != null
+    if socket != null && balloon
```

Comment:
This line shoud be changed in the other commit which probably should also be an extra commit.